table/src/main/java/com/bazaarvoice/emodb/table/db/astyanax/AstyanaxTableDAO.java

```diff
@@ -471,10 +534,10 @@ public void run(Runnable ignored) {
                     @Override
                     public void run(Runnable progress) {
 
-                        // Delay the final purge until we're confident we'll catch everything.
-                        if (iteration == 2) {
-                            checkPlacementConsistent(storage.getPlacement(), droppedAt);
-                        }
+                        // Delay the purge until we're confident we'll catch everything.
+                        checkPlacementConsistent(storage.getPlacement(), droppedAt);
+
+                        checkForAnyNonCompleteTableEvents(json.getTable(), storage.getUuidString(), TableEvent.Action.DROP);
 
                         purgeData(json, storage, iteration, progress);
```

Comment:
Big question - How are we blocking the actual purge if there are still events for the table with non-complete state? I'm missing something here.  From my understanding, Even though we are marking them as ready here - and in parallel the Table event processor would send those updates to megabus ref, it doesn't mean that the actual megabus topic will have the deletes before the purge here.   
   
When I reviewed this last time, I remember seeing something like relying on some exception and re-trying again. But not seeing that anywhere here. for sure I'm missing something. Please clarify. thanks!

Reply:
`checkForAnyNonCompleteTableEvents` will throw a `PendingTableEventsException` if there are pending events, which will prevent purge.